src/java/org/apache/cassandra/gms/FailureDetector.java

```diff
@@ -93,16 +91,12 @@ public FailureDetector()
 
     private static long getInitialValue()
     {
-        String newvalue = System.getProperty("cassandra.fd_initial_value_ms");
```

Comment:
I am not completely sure we can do this. If we move `Gossiper.intervalInMillis * 2` to `CassandraRelevantProperties`, is not it true that Gossiper's static variables / block will be initialized too? What are the consequences of doing that?

Consider this:

````
public class MyTest
{
    public static final String[] ARGS = {};

    @Test
    public void test()
    {
        MyTest.main(ARGS);
    }

    public static void main(String[] args)
    {
        System.out.println(Gossiper.b);
    }

    public static class Gossiper
    {
        static
        {
            System.out.println("in static initialiser of A");
        }

        public static int a = initA();

        public static int b = initB();

        public static int initA()
        {
            System.out.println("in A.initA");
            return 1;
        }

        public static int initB()
        {
            System.out.println("in A.initB");
            return 5;
        }
    }
}
````

This will print:

````
in static initialiser of A
in A.initA
in A.initB
5
````

This basically means that by calling `FD_INITIAL_VALUE_MS`, it will eagerly call static initializer in Gossiper too when that enum is being initialized. By doing that, we are changing the previous behavior.

I would rewrite this in such a way that `Gossiper.intervalInMillis` would be called only in case it has to be

````
    private static long getInitialValue()
    {
        long defaultValue = Gossiper.intervalInMillis * 2;
        String value = FD_INITIAL_VALUE_MS.getString();
        if (value == null)
            return defaultValue;

        long newValue = Long.parseLong(value);

        if (newValue != defaultValue)
            logger.info("Overriding {} from {}ms to {}ms",
                FD_INITIAL_VALUE_MS.getKey(), defaultValue, newValue);

        return newValue;
    }
````

Reply:
I reworked the code. 
Please take a look at it again.

I think it is safe for the initialization order of the static variables to move the `Gossiper.intervalInMillis * 2` to a constant variable, thus this part of the code will look much cleaner.